### PR TITLE
Updated the Contributing guidelines

### LINKED ISSUE #192: Contribution Guidelines out of sync between master and reorganise branches
The Contribution Guidelines on the `reorganise` branch are inaccurate (they reference the wrong branch). They should reflect the (more up to date) Contribution Guidelines on the `master` branch. 

Eventually they'll be merged into one, but this will prevent confusion in the interim.

### PR DESCRIPTION
**Pull request by heisenberg-2505**
Closes #192 .

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -34,14 +28,12 @@
 * Before you push your changes to GitHub, make sure that your code compiles and runs without any errors or warnings.
 
 ## C++ coding guidelines
-
-If you are contributing C++ code to this repo, make sure to read the [C++ Coding Guidelines](../C++/CODING_GUIDELINES.md).
+If you are contributing C++ code to this repo, make sure to read the [C++ Coding Guidelines](C++/CODING_GUIDELINES.md).
```
Comment (faheel) [inline .github/CONTRIBUTING.md]: The previous link was correct.

Comment (faheel) [reply .github/CONTRIBUTING.md]: Also, change it to a reference style link.

**Review by faheel — CHANGES_REQUESTED** [review]
Please change back the modified links to reference-style links (i.e. `[content to link][link-label]`). This helps to keep the text version more readable.

**Comment by faheel** [comment]
I just realised that since the final version will be almost exactly like the one in `master`, it would be simpler (and quicker) if I just include the commit in which I updated this file in `master` to `reorganise`.

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (heisenberg-2505) [reply .github/CONTRIBUTING.md]: I have made the changes.

**Pull request merged**